source/common/upstream/health_discovery_service.cc

```diff
@@ -203,6 +205,11 @@ void HdsDelegate::processMessage(
       cluster_config.add_health_checks()->MergeFrom(health_check);
     }
 
+    // Add transport_socket_match to cluster for use in host connections.
+    for (auto& transport_socket_match : cluster_health_check.transport_socket_matches()) {
+      cluster_config.add_transport_socket_matches()->MergeFrom(transport_socket_match);
```

Comment:
You might be able to get away with `cluster_config.mutable_transport_socket_matches()->MergeFrom(cluster_health_check);`, based on https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.repeated_field#RepeatedPtrField.

Reply:
Thanks! I was unsure if this would work at first on repeated fields, and decided to go with the implementation above for individual health checks. I went ahead and made this change because it is most certainly cleaner.